index.src.html

```diff
@@ -760,10 +760,10 @@ <h4 id="interfaces-credential-types-passwordcredential"><code>PasswordCredential
   :   <dfn attribute for="PasswordCredential">additionalData</dfn>
   ::  If the developer wishes to specify additional data to insert into the
       request body when submitting the credential information to a remote
-      endpoint, they can do so by assigning a {{FormData}} or
-      {{URLSearchParams}} object to this attribute. The credential information
-      will be mixed into the object to produce the body. The value is `null`
-      unless otherwise specified.
+      endpoint, they can do so by assigning a {{FormData}},
+      {{URLSearchParams}} or a generic object to this attribute. The
```

Comment:
> I think what you want is something similar to serializable types.

It seems like the web developer would be free to define the serializing method for their type. Is there a way to prevent this method from accessing the password value injected by the user agent?

Is serializability a language-independent counterpart of what JSON stringification is in JavaScript? If so, the Credential Manager API spec needs, in addition to requesting serializability, define how the user agent inserts the password value (and username). For a record, this would be defined by adding/replacing two mappings, with the keys defined by the developer, and values being the username and the password.

I wonder if instead of record<DOMString, any> the spec could require a record<DOMString, DOMString> -- the web developer would serialize their part of the data first (creating the DOMString values), the user agent would add two more such mappings, and then serialize the whole without complexity.

> No. it would be converted to an object.

Thanks, I misread the specification.